cmd/thanos/query.go

```diff
@@ -753,17 +763,25 @@ func engineFactory(
 		if ld < r {
 			lookbackDelta = time.Duration(r) * time.Millisecond
 		}
-		engines[i] = newEngine(promql.EngineOpts{
+
+		newEngineOpts := promql.EngineOpts{
 			Logger:                   eo.Logger,
 			Reg:                      wrapReg(i),
 			MaxSamples:               eo.MaxSamples,
 			Timeout:                  eo.Timeout,
-			ActiveQueryTracker:       eo.ActiveQueryTracker,
 			LookbackDelta:            lookbackDelta,
 			NoStepSubqueryIntervalFn: eo.NoStepSubqueryIntervalFn,
 			EnableAtModifier:         eo.EnableAtModifier,
 			EnableNegativeOffset:     eo.EnableNegativeOffset,
-		})
+		}
+		if activeQueryDir != "" {
+			resActiveQueryDir := filepath.Join(activeQueryDir, getActiveQueryDirBasedOnResolution(r))
+			newEngineOpts.ActiveQueryTracker = promql.NewActiveQueryTracker(resActiveQueryDir, maxConcurrentQueries, logger)
+		} else {
+			newEngineOpts.ActiveQueryTracker = eo.ActiveQueryTracker
```

Comment:
Do we need this `else` block? If `activeQueryDir` is empty string then the `ActiveQueryTracker` is just nil if I understand correctly

Reply:
@yeya24 in the current implementation, this line explicitly adds `eo.ActiveQueryTracker` https://github.com/thanos-io/thanos/blob/99239df3ba6ceb60ada032225c92622c5ee9ae4b/cmd/thanos/query.go#L761. 
I have attempted to retain that when I added the `else`.

Reply:
Let's remove it to avoid confusion.

Reply:
Or you could add a comment about it is nil.

Reply:
Thanks, done.
If possible could you please retrigger the CI tests once again?
Thanks!